File: communication/mosquitto/simple-demo/go-server/publisher/handler.go
package publisher

import (
	"encoding/json"
	"net/http"
	"time"

	"github.com/labstack/echo/v4"

	"github.com/seungyeop-lee/poc/communication/mosquitto/simple-demo/go-server/common"
	"github.com/seungyeop-lee/poc/communication/mosquitto/simple-demo/go-server/config"
)

type Handler struct {
	mqtt *config.MQTT
}

func NewHandler(mqtt *config.MQTT) *Handler {
	return &Handler{mqtt: mqtt}
}

func (h *Handler) Apply(e *echo.Echo) {
	e.POST("/publish", h.PublishMessage)
}

func (h *Handler) PublishMessage(c echo.Context) error {
	payload := common.PayloadMessage{
		Name:        "from go server",
		CurrentTime: time.Now().UnixMilli(),
		Enable:      true,
		Data:        []byte("Hello from Go MQTT!"),
	}

	body, err := json.Marshal(payload)
	if err != nil {
		return err
	}

	topic := common.TopicPrefix + "go"
	token := h.mqtt.Client.Publish(topic, common.QoS, false, body)
	if token.Wait() && token.Error() != nil {
		return token.Error()
	}

	return c.NoContent(http.StatusOK)
}

File: communication/mosquitto/simple-demo/python-server/mosquitto_python_server/publisher/handler.py
import base64
import json
import time

from fastapi import APIRouter

from mosquitto_python_server.common.types import PayloadMessage
from mosquitto_python_server.common.values import TOPIC_PREFIX, QOS
from mosquitto_python_server.config.mqtt import MQTT

router = APIRouter()


class Handler:
    def __init__(self, mqtt: MQTT):
        self.mqtt = mqtt
        self.router = router

    def setup_routes(self):
        self.router.add_api_route("/publish", self.publish_message, methods=["POST"])

    async def publish_message(self):
        payload = PayloadMessage(
            name="from python server",
            currentTime=int(time.time() * 1000),
            enable=True,
            data=base64.b64encode(b"Hello from Python MQTT!").decode('utf-8')
        )

        message = json.dumps(payload.model_dump(by_alias=True))
        topic = TOPIC_PREFIX + "python"
        self.mqtt.client.publish(topic, message, QOS)

        return {"status": "ok"}
